components/brave_wallet/browser/internal/hd_key_ed25519.cc

```diff
@@ -62,74 +80,61 @@ std::string HDKeyEd25519::GetPath() const {
 // https://github.com/satoshilabs/slips/blob/master/slip-0010.md#private-parent-key--private-child-key
 std::unique_ptr<HDKeyEd25519> HDKeyEd25519::DeriveHardenedChild(
     uint32_t index) {
-  auto child_private_key = private_key_->unwrap().derive_hardened_child(index);
-  if (!child_private_key->is_ok()) {
-    VLOG(0) << std::string(child_private_key->error_message());
+  if (index >= kHardenedOffset) {
     return nullptr;
   }
-  auto child_path =
-      base::StrCat({path_, "/", base::NumberToString(index), "'"});
-  return std::make_unique<HDKeyEd25519>(std::move(child_path),
-                                        std::move(child_private_key));
-}
+  std::array<uint8_t, 37> hmac_payload = {};
 
-std::unique_ptr<HDKeyEd25519> HDKeyEd25519::DeriveChildFromPath(
-    const std::string& path) {
-  if (path_ != kMasterNode) {
-    VLOG(0) << "must derive only from master key";
-    return nullptr;
-  }
+  auto span_writer = base::SpanWriter(base::span(hmac_payload));
+  // https://github.com/satoshilabs/slips/blob/master/slip-0010.md#private-parent-key--private-child-key
+  span_writer.WriteU8BigEndian(0);
+  span_writer.Write(GetPrivateKeyAsSpan());
+  span_writer.WriteU32BigEndian(index + kHardenedOffset);
+  DCHECK_EQ(span_writer.remaining(), 0u);
 
-  auto child_private_key = private_key_->unwrap().derive(path);
-  if (!child_private_key->is_ok()) {
-    VLOG(0) << std::string(child_private_key->error_message());
-    return nullptr;
-  }
+  return DeriveFromHmacPayload(chain_code_, hmac_payload);
```

Comment:
Where is the following check in number 5 of [this section](https://github.com/satoshilabs/slips/blob/master/slip-0010.md#private-parent-key--private-child-key) being done?

```
If parse256(IL) ≥ n or parse256(IL) + kpar (mod n) = 0 (resulting key is invalid):
	let I = HMAC-SHA512(Key = cpar, Data = 0x01 || IR || ser32(i) 		and restart at step 2.
```

It seems that it may need to be done in the function called here or in `DeriveKeyPairFromPrivateKey` since the call to [BoringSSL isn't](https://boringssl.googlesource.com/boringssl/+/refs/heads/master/crypto/curve25519/curve25519.cc#1970) doing this check.

Reply:
We stop at step 4
`If curve is ed25519 or curve25519: The returned child key ki is IL.`

Reply:
Ahh makes sense, my mistake.